File: mobile/src/main/java/fr/poker/stibriana/stibrianapoker/MainActivity.java
package fr.poker.stibriana.stibrianapoker;

import android.app.Activity;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.KeyEvent;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;


public class MainActivity extends Activity {

    int points;
    int gain;
    double coef[] = {0, 5, 4, 3, 2.5, 2.25, 2, 1.75, 1.5, 1.25};
    double coef2[] = {0, 3.5, 2, 7 / 6.5};
    int joueurs = 0;
    int place = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        EditText joueursRecup = (EditText) findViewById(R.id.joueurs);
        EditText placeRecup = (EditText) findViewById(R.id.place);

        joueursRecup.setOnKeyListener(new View.OnKeyListener() {
            @Override
            public boolean onKey(View view, int actionId, KeyEvent event) {
                if (KeyEvent.KEYCODE_ENTER == actionId) {
                    sending();
                }
                return false;
            }
        });

        placeRecup.setOnKeyListener(new View.OnKeyListener() {
            @Override
            public boolean onKey(View view, int actionId, KeyEvent event) {
                if (KeyEvent.KEYCODE_ENTER == actionId) {
                    sending();
                }
                return false;
            }
        });
    }


    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }

    public void sendMessage(View view) {

        sending();

    }

    public void sending(){
        EditText joueursRecup = (EditText) findViewById(R.id.joueurs);
        EditText placeRecup = (EditText) findViewById(R.id.place);
        TextView Pointsdisplay = (TextView) findViewById(R.id.textView);
        TextView Gaindisplay = (TextView) findViewById(R.id.textView2);

        if (joueursRecup.getText().toString().trim().length() == 0) {
            Toast.makeText(this, "Veuillez entrez le nombre de joueurs", Toast.LENGTH_SHORT).show();
        } else if (placeRecup.getText().toString().trim().length() == 0) {
            Toast.makeText(this, "Veuillez entrez votre place", Toast.LENGTH_SHORT).show();
        } else {

            joueurs = Integer.valueOf(joueursRecup.getText().toString());
            place = Integer.valueOf(placeRecup.getText().toString());

            if (place > joueurs) {
                Toast.makeText(this, "Vous avez rentré de mauvaises informations", Toast.LENGTH_SHORT).show();
            } else {

                if (place > 9) {
                    points = (int) Math.round((joueurs + 1 - place));
                } else {
                    points = (int) Math.round(coef[place] * (joueurs - place + 1));
                }

                if (place < 4) {
                    gain = (int) Math.round((joueurs * coef2[place]));
                } else if (place == 4) {
                    gain = joueurs * 7 - (int) Math.round(joueurs * 3.5) - (int) Math.round(joueurs * 2) - (int) Math.round(joueurs * 7 / 6.5);
                } else {
                    gain = 0;
                }

                Pointsdisplay.setText(String.valueOf(points) + "pts");
                Gaindisplay.setText(String.valueOf(gain) + "€");
            }
        }
    }

    public void goScore(View view) {
        Uri uri = Uri.parse("https://docs.google.com/spreadsheets/d/1J64YMpKCAftDtn9-YyFMVYranCEjTOKGyP9nzkgpY4I/edit?usp=sharing");
        Intent intent = new Intent(Intent.ACTION_VIEW, uri);
        startActivity(intent);
    }
}

File: mobile/src/main/java/fr/poker/stibriana/stibrianapoker/MainActivity.kt
package fr.poker.stibriana.stibrianapoker

import android.app.Activity
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.KeyEvent
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.EditText
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*
import kotlin.math.roundToInt

const val LINK_SCORE_SHEET = "https://docs.google.com/spreadsheets/d/1hdUt6IjVTEQW_RgmaxluXW4IRBAMucE0MBxc-ypa9pg/edit?usp=sharing"

class MainActivity : Activity() {

    private var points: Int = 0
    private var gain: Int = 0
    private var coef = doubleArrayOf(0.0, 5.0, 4.0, 3.0, 2.5, 2.25, 2.0, 1.75, 1.5, 1.25)
    private var coef2 = doubleArrayOf(0.0, 3.5, 2.0, 7 / 6.5)
    private var playersNumber = 0
    private var placeNumber = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val players = findViewById<View>(R.id.players) as EditText
        val place = findViewById<View>(R.id.place_editext) as EditText

        players.setOnKeyListener { _, actionId, event ->
            if (KeyEvent.KEYCODE_ENTER == actionId) {
                sending()
            }
            false
        }

        place.setOnKeyListener { _, actionId, _ ->
            if (KeyEvent.KEYCODE_ENTER == actionId) {
                sending()
            }
            false
        }

        send_result.setOnClickListener {
            sending()
        }

        goToScore.setOnClickListener {
            goScore()
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        val id = item.itemId

        return if (id == R.id.action_settings) {
            true
        } else super.onOptionsItemSelected(item)

    }

    private fun sending() {
        if (players.text.toString().trim { it <= ' ' }.isEmpty()) {
            Toast.makeText(this, getString(R.string.enter_players_number_text), Toast.LENGTH_SHORT).show()
        } else if (place_editext.text.toString().trim { it <= ' ' }.isEmpty()) {
            Toast.makeText(this, getString(R.string.enter_place_text), Toast.LENGTH_SHORT).show()
        } else {

            playersNumber = Integer.valueOf(players.text.toString())
            placeNumber = Integer.valueOf(place_editext.text.toString())

            if (placeNumber > playersNumber) {
                Toast.makeText(this, getString(R.string.wrong_informaiton_message_text), Toast.LENGTH_SHORT).show()
            } else {

                points = if (placeNumber > 9) {
                    (playersNumber + 1 - placeNumber).toFloat().roundToInt()
                } else {
                    (coef[placeNumber] * (playersNumber - placeNumber + 1)).roundToInt()
                }

                gain = if (placeNumber < 4) {
                    (playersNumber * coef2[placeNumber]).roundToInt()
                } else if (placeNumber == 4) {
                    playersNumber * 7 - (playersNumber * 3.5).roundToInt() - (playersNumber * 2).toFloat().roundToInt() - (playersNumber * 7 / 6.5).roundToInt()
                } else {
                    0
                }

                points_txt.text = getString(R.string.showed_pts, points)
                gain_txt.text = getString(R.string.showed_gain, gain)
            }
        }
    }

    private fun goScore() {
        val uri = Uri.parse(LINK_SCORE_SHEET)
        val intent = Intent(Intent.ACTION_VIEW, uri)
        startActivity(intent)
    }
}

